src/arrayMethodSort.js

```diff
@@ -4,8 +4,86 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  // Define the custom sort2 method on the Array prototype
+  [].__proto__.sort2 = function (compareFunction) {
+    const arr = this; // Reference to the current array
+    const len = arr.length; // Length of the array
+
+    /**
+     * Default comparison function (lexicographical order)
+     * Converts elements to strings and compares them.
+     * @param {*} a - First element to compare
+     * @param {*} b - Second element to compare
+     * @returns {number} - Negative if a < b, positive if a > b, zero if equal
+     */
+    const defaultCompare = (a, b) => {
+      const aStr = String(a);
+      const bStr = String(b);
+
+      if (aStr < bStr) {
+        return -1;
+      }
+
+      if (aStr > bStr) {
+        return 1;
+      }
+
+      return 0;
+    };
+
+    // Check if compareFunction is provided and is a function
+    if (
+      compareFunction !== undefined &&
+      typeof compareFunction !== 'function'
+    ) {
+      throw new TypeError('compareFunction must be a function');
+    }
+
+    // Use the provided compareFunction or the default one
+    const compare =
+      typeof compareFunction === 'function' ? compareFunction : defaultCompare;
+
+    /**
+     * In-place Quicksort implementation
+     * @param {number} start - Starting index of the segment to sort
+     * @param {number} end - Ending index of the segment to sort
+     */
```

Comment:
I see that you have a lot of empty lines in your code. While it's good to have some empty lines to separate logical blocks of code, having too many can make your code look messy. Try to remove unnecessary empty lines.